device/dhcp/clients/v4/dhcpv4client.yaml

```diff
@@ -0,0 +1,60 @@
+components:
+  schemas:
+    Device.Dhcpv4client:
+      description: >-
+        Configuration for emulated DHCPv4 Client on a single Interface.
+        https://www.rfc-editor.org/rfc/rfc2131.html
+      type: object
+      x-status:
+        status: under_review
+      required: [name]
+      properties:
+        name:
+          x-include: ../../../../common/common.yaml#/components/schemas/Named.Object/properties/name
+          x-field-uid: 1
+        choice:
+          description: >-
+            The client receives one or more DHCPOFFER messages from one or more servers and client may choose to wait for multiple responses.
+            The client chooses one server from which to request configuration
+            parameters, based on the configuration parameters offered in the DHCPOFFER messages.
+            first_server: if selected, the subnet accepts the IP addresses offered by the first server to respond with an offer of IP addresses.
+            server_address: The address of the DHCP server from which the subnet will accept IP addresses.
+            If server_address is selected then next field 'server_address' to be assigned with a IP address.
+          type: string
+          default: first_server
+          x-field-uid: 2
+          x-enum:
+            first_server:
+              x-field-uid: 1
+            server_address:
+              x-field-uid: 2
+        first_server:
+          type: boolean
+          default: true
+          x-field-uid: 3
+        server_address:
+          $ref: '#/components/schemas/Dhcpv4client.Server'
```

Comment:
You can directly set the type as ip. I dont think a nested object is needed for this since it has a single attribute.

Reply:
yeah